TST: fix #6542, add tests to check non-iterable argument raises in hstack and related functions.

gkBCCN opened the pull request:
Collaboration with @clauslang, @ClaudiaWinklmayr, @zukkini:

-test that calls to hstack, vstack, stack, hsplit, vsplit, dsplit, dstack with non-iterable inputs raise errors

ahaldane committed:
MAINT: Hide nan warnings for internal ma computations

ahaldane committed:
TST: Fix some MA tests to avoid looking at the .data attribute

The MaskedArray.data attribute is unreliable for tests because it can
contain arbitraty junk data at masked positions. Instead, all
MaskedArray tests should look at marr.filled(0) to check if we got the
same result.

dongjoon-hyun committed:
DOC: Fix more typos in docs and comments.

jakirkham committed:
TST: Install `pytz` in the CI.

This should allow for some tests that require `pytz` to run during CI.

gkBCCN committed:
TST: Fix #6542: Add test for non-iterable input...

...to hstack, vstack, and stack

gkBCCN committed:
TST: Fix #6542: Add test for non-iterable input...

...to hstack, vstack, and stack

simongibbons committed:
DOC: Fixed math rendering in tensordot docs.

Fixes #7339

gkBCCN committed:
TST: Fix #6542: Add test for non-iterable input...

...for hsplit, vsplit, dsplit, dstack

gkBCCN committed:
TST: Fix #6542: Add test for non-iterable input...

...for hsplit, vsplit, dsplit, dstack

charris committed:
Merge pull request #7338 from jakirkham/use_pytz_ci

TST: Install `pytz` in the CI.

njsmith committed:
Merge pull request #7340 from simongibbons/tensordot_doc_fix

DOC: Fixed math rendering in tensordot docs.

njsmith committed:
Merge pull request #7344 from dongjoon-hyun/fix_typos

DOC: Fix more typos in docs and comments.

gkBCCN commented:
Changed title to add hash to ticket number so that it links correctly.

madphysicist committed:
Maint: Removed extra space from `ureduce` 

This is just me being OCD. I am not sure this even merits a full commit, much less a PR, but here goes anyway.

charris committed:
Merge pull request #7348 from madphysicist/patch-1

Maint: Removed extra space from `ureduce`

charris commented on numpy/lib/tests/test_shape_base.py:
```diff
@@ -238,6 +248,17 @@ class TestVsplit(TestCase):
+
+    def test_0D_array(self):
+        a = np.array(1)
+        try:
```
This doesn't look right, what are you testing? Why catch the exception? Same below.

charris committed:
Merge pull request #7351 from ahaldane/ma_fix_1_datatest

TST: Fix some MA tests to avoid looking at the .data attribute

charris committed:
Merge pull request #7349 from ahaldane/ma_fix_3_ignorenan

MAINT: Hide nan warnings for masked internal MA computations

chiffa committed:
BUG: constant padding expected wrong type in constant_values

Constant padding on 4 sides of a 2d array expected a numpy ndarray, and not a ndarray like (tuple, list, ...)
Detailed description is in the issue  #7353

chiffa committed:
TST: added a test for constant padding on 4 sides of a 2d array

This test exposes padding bug described in the issue #7353

njsmith committed:
Merge pull request #7358 from chiffa/master

BUG: pull request related to the issue #7353

gkBCCN committed:
TST: Fix #6542: TestVsplit, TestDsplit updated...

...use assert_raises() instead of catching the exception itself.

gkBCCN committed:
Merge branch 'bug-fix-6542' ...

...of https://github.com/gkBCCN/numpy into bug-fix-6542

Conflicts:
	numpy/lib/tests/test_shape_base.py

charris commented:
I think you need to start over here. First, you are working off master, which is a no-no and probably why you now have 21 commits in this PR. Make a branch from master in work in that. It is possible that your master branch is now out of sync with the numpy repo, so you might want redo you own repo. You can also `git reset --hard <hash-before-your-work>` and then pull from upstream (the numpy repo) to fix that, but it requires a bit more git foo.

I'm going to close this. Please make another PR.
